frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -115,21 +124,25 @@ function ItemPicker<TId>({
 
     const collectionItems = list
       .filter(canWriteToCollectionOrChildren)
+      .filter(
```

Comment:
Does this filter statement need to be recursive, like `canWriteToCollectionOrChildren`?

Reply:
It doesn't need to be, the purpose of this new `filter` is to decide whether we want to render the immediate collections in the current collection. And that information could be computed without recursively traversing the whole collection tree.